how-to/misc/identity-provider-enablement.rst

```diff
@@ -0,0 +1,354 @@
+
+There are three mandatory configuration parameters and one optional parameter:
+
+* `client_id` - mandatory
+* `client_secret` - mandatory
+* `microsoft-tenant` - mandatory
+* `label` - optional
+
+Example config:
+
+::
+
+    client-id: client_id_obtained_from_your_console
+    client-secret: client_secret_associated_with_the_id
+    microsoft-tenant: tenant-uuid-goes-here
+    label: "Log in with Entra ID"
+
+Adding an external IDP
+~~~~~~~~~~~~~~~~~~~~~~
+
+Adding an external IDP is similar to adding a Canonical Identity Platform provider:
+
+::
+
+    sunbeam identity provider add \
+        google openid my-google-idp \
+        --config google.yaml
+
+
+Now we can list the providers:
+
+::
+
+    sunbeam identity provider list
+    +-----------------------------+------------+----------+
+    | Name                        | Provider   | Protocol |
+    +-----------------------------+------------+----------+
+    | Keystone Credentials        | Built-in   | keystone |
+    | canonical-identity-platform | canonical  | openid   |
+    | my-google-idp               | google     | openid   |
+    +-----------------------------+------------+----------+
+
+Okta and Entra ID have identical procedures.
+
+Make a note of the name of the provider and of the protocol. We will use them in the next steps to enable these providers in keystone.
+
+Note, you should already see them in `Horizon`, but you will only be able to use them after we've mapped them to domains and projects. Examples bellow.
+
+Removing a provider
+~~~~~~~~~~~~~~~~~~~
+
+Removing a provider is a matter of running:
+
+::
+
+    sunbeam identity provider remove my-google-idp --yes-i-mean-it
+
+
+Note, this will not remove any resources created by the cloud administrator using the `openstack` command.
+
+Making use of the new providers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that we've made the providers available to the cloud, we can enable them in keystone, map them to a domain and create rules on how users should
+be mapped to projects.
+
+You can create a new domain or you can use an existing domain to map it to the IDP. For the purposes of this guide, we'll create a new one:
+
+::
+
+    openstack domain create \
+        --description="Federated Google domain" \
+        google
+
+Get the issuer URL for the desired IDP. In this case we'll go with `my-google-idp` from the output above:
+
+::
+
+    ISSUER_URL=$(sunbeam identity provider list \
+        --format=yaml |  yq -r '."my-google-idp".issuer_url')
+
+Create the identity provider in Keystone:
+
+::
+
+    openstack identity provider create \
+        --remote-id $ISSUER_URL \
+        --domain google \
+        my-google-idp
+
+
+Note, the name of the identity provider must match the name in the table outputted by sunbeam.
+
+Create a group which we will assign to federated users:
+
+::
+
+    openstack group create federated_users \
+        --domain google
+
+Create a project. You can of course name something else:
```

Comment:
A bit of a nit, but in the second part of this maybe something more like:

The following example creates a project named `federated_project`.

I think its safe to assume that the user will be able to adjust the name of the project on their own and the suggestion to name it something else is unnecessary. A slight tweak to the wording makes it clear the user can select their own by simply explaining what you're doing here.